clash-ghc/src-ghc/Clash/GHC/NetlistTypes.hs

```diff
@@ -265,13 +265,15 @@ liftE
   -> ExceptT e (MaybeT m) a
 liftE = mapExceptT (MaybeT . pure . Just . coerce)
 
-domTag :: Monad m => Type -> ExceptT String (MaybeT m) String
-domTag (LitTy (SymTy tag)) = pure tag
-domTag ty = throwE $ "Can't translate domain tag" ++ showPpr ty
+domTag :: Monad m => TyConMap -> Type -> ExceptT String (MaybeT m) String
+domTag m (coreView1 m -> Just ty) = domTag m ty
+domTag _ (LitTy (SymTy tag)) = pure tag
+domTag _ ty = throwE $ "Can't translate domain tag" ++ showPpr ty
```

Comment:
We shouldn't be seeing this error anymore hopefully, but it would be nice to add ": "
```haskell
domTag _ ty = throwE $ "Can't translate domain tag: " ++ showPpr ty
```